### PR TITLE
Added ESLint and fixed the errors it found

### PR DESCRIPTION
**Pull request by mattbishop**
Also fixes typo from https://github.com/prayerslayer/js.spec/issues/17

I didn't include it into the build process directly, so you have to use `yarn lint` to run it. Some people don't like lint running all the time.

### CODE REVIEW

File: lib/regex.js [hunk 1]
```diff
@@ -87,9 +87,13 @@ export function deriv(regex, x) {
       return altDeriv(regex, x)
     case cat:
       return catDeriv(regex, x)
+    case amp:
```
Comment (mattbishop) [inline lib/regex.js]: fixes `no-unused-vars` check.

**Comment by mattbishop** [comment]
All right, I'll add a test then.

**Comment by mattbishop** [comment]
Interesting, I am adding a new test for registry.js as it has the least amount of coverage. I wrote a test for `_clear` and now the map.test is failing in the alias tests. I suppose the test is relying on previous tests setting up the registry with mappings.

```      works on spec aliases
        1) [happy case]
        2) [invalid nested key]
        3) [missing required nested key]
        ✓ [wrong required nested key]

**Comment by mattbishop** [comment]
I have figured out what is going wrong. Mocha is running map.test..."works on spec aliases" which calls generateConformTests() and that method generates a bunch it it() calls with tests; however they are not run right then. They are saved for some time later.

Specifically, they are run _after_ my new registry._clear() test blows away all the symbols in the registry.

**Comment by mattbishop** [comment]
Hmm failed CI due to 504 in coverall?

**Comment by mattbishop** [comment]
test this please

**Comment by mattbishop** [comment]
The above error is a Gateway Timeout:

The web server reported a gateway time-out error.

Ray ID: 3404423cfd7e2474
Your IP address: 54.87.185.35
Error reference number: 504
Cloudflare Location: Ashburn

**Comment by mattbishop** [comment]
Might be this: https://www.traviscistatus.com/incidents/3s7jqhhzt8k1

@prayerslayer would you be able to enable 'test this please' support in Travis? I want to rerun the build when they think their connectivity issues are fixed but it feels like a hack to have to push a whitespace change to trigger a rebuild.

**Comment by prayerslayer** [comment]
> would you be able to enable 'test this please' support in Travis?

I don't know how to do it... Your effort is appreciated, but is this urgent? We could also wait until the incident is over.

**Comment by mattbishop** [comment]
Not urgent I just have a short attention span ;)

**Comment by scotttrinh** [comment]
@mattbishop I'm not sure if Travis will do this, but my company's Jenkins CI bot re-checks if you close and reopen.

**Comment by mattbishop** [comment]
Thanks Scott I'll try that next week once Travis <-> Coveralls connectivity is stable.

**Comment by mattbishop** [comment]
Both Coveralls and Travis report good health so I'll try closing and reopening this PR to trigger a new build.

**Comment by mattbishop** [comment]
...and open again.

**Comment by prayerslayer** [comment]
Thanks for your efforts, I'll merge and tweak the eslint config a little :)

**Pull request merged**